packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,242 @@
+            quit: this.quitHandler.bind(this),
+            stop: this.stopHandler.bind(this),
+            start: this.startHandler.bind(this),
+        };
+    }
+
+    apply(compiler) {
+        if (!isWebpack5) {
+            logger.error('Interactive is not supported on webpack v4 and less');
+            process.exit(1);
+        }
+
+        // Configure stdin for keypress event
+        const stdin = process.stdin;
+        stdin.setEncoding('utf-8');
+        stdin.setRawMode(true);
+        readline.emitKeypressEvents(stdin);
+
+        // Configure keypress event for actions
+        const actions = Object.keys(this.keys);
+        stdin.on('keypress', (_, actionKey) => {
+            const possibleActions = actions.filter((action) => {
+                return this.keys[action] === actionKey.name;
+            });
+
+            if (possibleActions.length === 0) {
+                return;
+            }
+
+            if (possibleActions.length > 1) {
+                throw new Error('Multiple actions are provided for same key');
+            }
+
+            const action = possibleActions[0];
+            this.handlers[action](compiler);
+        });
+
+        if (compiler.compilers) {
+            this.isMultiCompiler = true;
+            this.compilers = compiler.compilers;
+        }
+
+        // Register Custom Hook for printing after clrscr
+        if (!compiler.hooks.beforeInteractiveStats) {
+            compiler.hooks = {
+                ...compiler.hooks,
+                beforeInteractiveStats: new SyncHook(),
+            };
+        }
+
+        // Clear for first run as well
+        clrscr();
+        compiler.hooks.beforeInteractiveStats.call();
+
+        if (!this.isMultiCompiler) {
+            // Clear output on watch invalidate
+            compiler.hooks.beforeCompile.tap(this.name, () => {
+                clrscr();
+                compiler.hooks.beforeInteractiveStats.call();
+            });
+
+            compiler.hooks.afterDone.tap(this.name, () => {
+                setTimeout(() => {
+                    console.log('\n\n');
+                    spawnCommand('compilation completed', true);
+                }, 1);
+            });
+        } else {
+            const helperPlugin = new InteractiveModeMultiCompilerHelperPlugin();
+
+            // Register helper plugin on each of child compiler
+            for (const childCompiler of this.compilers) {
+                helperPlugin.apply(childCompiler);
+            }
+
+            compiler.hooks.done.tap(this.name, () => {
+                const allDone = this.compilers.reduce((result, childCompiler) => {
+                    return result && !childCompiler.watching.running;
+                }, true);
+
+                if (!allDone) return;
+
+                setTimeout(() => {
+                    console.log('\n\n');
+                    spawnCommand('all compilations completed', true);
+                }, 100);
+            });
```

Comment:
Why we need `setTimeout` here? if we need wait v8 look we can use `process.nextTick(() => {})`

Reply:
Thanks, didn't knew about about nextTick. Updated accordingly.